README.md

```diff
@@ -145,6 +145,18 @@ trainer = Trainer(
 trainer.fit(model)
 ```
 
+## Using S3 Versioning to Manage Checkpoints
+When working with model checkpoints, users can accidentally overwrite or delete critical files. This can happen due to incorrect configuration, multiple hosts accessing the same storage path concurrently, or race conditions caused by usage of the async checkpoint interface.
+Amazon S3 bucket versioning provides a way to protect checkpoints from inadvertent loss. With versioning enabled, S3 preserves every version of every object in the bucket.
+
+When versioning is enabled on an S3 bucket:
+* S3 stores a version history of your objects. The current version is returned by default.
+* You can retrieve previous versions of your objects.
+* Unintended overwrites and deletions are reversible.
```

Comment:
```suggestion
When versioning is enabled on an S3 bucket, deletions insert a delete marker instead of removing the object permanently. The delete marker becomes the current object version. If you overwrite an object, it results in a new object version in the bucket. You can always restore the previous version. See [Deleting object versions from a versioning-enabled bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/DeletingObjectVersions.html) for more details on managing object versions.
```